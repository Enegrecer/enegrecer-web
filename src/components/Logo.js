import React from 'react';
import logo from './menu/images/logo.png';
const Logo = () => (
  <div className="en-logo">
    <img src={logo}/>
  </div>
);

export default Logo;
