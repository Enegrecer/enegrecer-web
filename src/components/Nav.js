import React from 'react';
import { withHistoryPropTypes } from './sharedPropTypes';

const Nav = () => (
    <ul className='right hide-on-med-and-down'>
      <li><a name="sobre" href="/sobre">Sobre</a></li>
    </ul>
);

Nav.propTypes = withHistoryPropTypes();

export default Nav;
