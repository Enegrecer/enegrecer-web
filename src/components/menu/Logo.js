import React from 'react';
import logo from './images/logo.png';
const Logo = () => (
  <div className="en-logo">
    <img src={logo} alt="Logo Verdade Seja Dita"/>
  </div>
);

export default Logo;
