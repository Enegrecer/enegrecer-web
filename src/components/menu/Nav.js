import React from 'react';

const Nav = () => (
    <ul className='right hide-on-med-and-down'>
      <li><a name="sobre" href="/sobre">Sobre</a></li>
    </ul>
);

export default Nav;
