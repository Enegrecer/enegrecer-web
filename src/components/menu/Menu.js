import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import logo from '../../imagens/logo.svg';
import './menu.css';

export default class Menu extends Component {
  componentWillMount() {
  }

  render() {
    return (
      <div className="col s12 menu">
        <div className="logo">
          <Link to="/">
            <img src={logo} alt="Logo Verdade Seja Dita" />
          </Link>
        </div>
        <div className="links">
          <Link to="/sobre">SOBRE</Link>
          <Link to="/depois-denuncia">E DEPOIS DA DENÃšNCIA?</Link>
          <Link to="/coletivo-enegrecer">O COLETIVO ENEGRECER</Link>
          <Link to="/denunciar">DENUNCIAR</Link>
        </div>
      </div>
    );
  }
}
