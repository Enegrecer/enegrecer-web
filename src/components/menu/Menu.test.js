import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Menu from './Menu';
import Logo from './../Logo';
import Nav from './../Nav';

describe('Menu', () => {
  const historyMock = { push: jest.fn() };
  const wrapper = shallow(<Menu history={historyMock} />);

  it('possui a mesma estrutura', () => {
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
  });

  it('tem um topbar', () => {
    expect(wrapper.find('Topbar').exists()).toBeTruthy();
  });

  it('tem um topbar com a logo na esquerda', () => {
    expect(wrapper.find('Topbar').props().leftComponent).toEqual(<Logo />);
  });

  it('tem um topbar com uma nav na direita com rota definida', () => {
    expect(wrapper.find('Topbar').props().rightComponent).toEqual(<Nav history={historyMock} />);
  });
});
