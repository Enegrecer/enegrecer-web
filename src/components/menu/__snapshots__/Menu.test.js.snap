// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu deve construir a pagina corretamente  1`] = `
<div
  className="col s12 menu"
>
  <div
    className="logo"
  >
    <Link
      replace={false}
      to="/"
    >
      <img
        alt="Logo Verdade Seja Dita"
        src="logo.svg"
      />
    </Link>
  </div>
  <div
    className="links"
  >
    <Link
      replace={false}
      to="/sobre"
    >
      SOBRE
    </Link>
    <Link
      replace={false}
      to="/depois-denuncia"
    >
      E DEPOIS DA DENÃšNCIA?
    </Link>
    <Link
      replace={false}
      to="/coletivo-enegrecer"
    >
      O COLETIVO ENEGRECER
    </Link>
    <Link
      replace={false}
      to="/denunciar"
    >
      DENUNCIAR
    </Link>
  </div>
</div>
`;
